### DISCUSSION TITLE
I lost my document 😭😭

### DISCUSSION THREAD

**Discussion by Aragorn42** [typst/typst#952] [Q&A]
1. Last night I wrote down a part of my document on typst's web app, and close my computer down, but today when I view it again I find that it's haven't saved down😣😣, I'm so sad and I wonder if it's because I use the Chinese mainland network? 
2. I used offline typst on vscode originally, but it can't run my code for some reason I don't know after I typed about 50 lines. If I can solve this problem, I can use vscode to manage my code's edition so there won't have the problem 1.
(My English is quite poor, I hope someone can understand my problem)

**Comment by Aragorn42** [comment]
PS:For the problem 1, it isn't not saved completely, it is the edition I edited an hour ago before I finished my work last night.

**Comment by szdytom** [comment]
for P1, no network problem for me in Guangzhou.

for P2, you can consider disable the built-in auto build from VS Code extension and manually do a `typst watch xx.typ` instead. There are indeed so many bugs in the LSP.

**Comment by Enivex** [comment]
In general it's a good idea to back up your work locally when using the web app (it even says so top right). It's still very early, and it's a question of when, not if, some bug will rear its head.
